frontend/src/components/Modal.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { Card, Button } from '.';
+import Warning from './Warning';
+
+export default function Modal(props: {
+  isOpen: boolean;
+  toggleModal: (value: boolean) => void;
+  deleteCategory: () => void;
+}) {
+  return (
+    <div
+      className={`${
+        props.isOpen ? 'visible' : 'invisible'
+      } z-50 fixed inset-0 bg-gray-600 bg-opacity-25 overflow-y-auto h-full w-full flex flex-col items-center justify-center`}
+    >
+      <div className=''>
+        <Card className='flex bg-white flex flex-col items-center justify-around p-12 h-96 rounded-2xl'>
+          <Warning className='w-32 text-red-300' />
+          <div>
+            <h1 className='text-3xl font-semibold'>
+              Are you sure you want to delete this category?
```

Comment:
Modal is a very generic component. So this message should be dynamic, so you can reuse it for next time smoothly.